### PR TITLE
feat: labels on proposals

### PR DESCRIPTION
**Pull request by ChaituVR**
Towards https://github.com/snapshot-labs/workflow/issues/143

### Summary:
- Allows labels in proposals
- Add tests to check if hasedTypes exist in hashedTypes.json (makes it easy to modify types in future)

### How to test:
- Run `yarn test`
- Edit test/examples/proposal.json with different label values

### CODE REVIEW

**Review by bonustrack — COMMENTED** [review]
Can we enforce a max lenght for label string? It can be 24

**Comment by ChaituVR** [comment]
> Can we enforce a max lenght for label string? It can be 24

Using `"maxLength": 8`, reason:
https://github.com/snapshot-labs/sx-monorepo/pull/829#discussion_r1786105787
https://github.com/snapshot-labs/snapshot.js/blob/master/src/schemas/space.json#L407

File: src/schemas/update-proposal.json [hunk 1]
```diff
@@ -40,6 +40,14 @@
             "turbo": 1000
           }
         },
+        "labels": {
```
Comment (wa0x6e) [inline src/schemas/update-proposal.json:43]: No unique items and pattern check on update ?

Comment (ChaituVR) [reply src/schemas/update-proposal.json:43]: opps missed it, added it now

**Review by wa0x6e — APPROVED** [review]
utAck

**Pull request merged**